omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java

```diff
@@ -263,7 +263,22 @@ public FragmentActionResult submit(UiSessionContext sessionContext, @RequestPara
             }
         }
 
-        InfoErrorMessageUtil.flashInfoMessage(request.getSession(), ui.message("registrationapp.createdPatientMessage", patient.getPersonName()));
+        // Ensure name is HTML-safe to prevent XSS in info messages
```

Comment:
Can you just a single encodeHtml call to the return value of `ui.message("registrationapp.createdPatientMessage", patient.getPersonName())`?

Reply:
That could work. The one drawback is that if the `ui.message()` function ever returns any kind of html-formatted text (e.g. `<b>` or `<span>` tags), that text will get html-encoded and will not work correctly.

Reply:
I see, then you need to get to do this;

`ui.message("registrationapp.createdPatientMessage", ui.encodeHtml(ui.format(patient.getPersonName())));`

The key thing is to leave the 'stringfying' of the person name object to the internals of the framework.

Reply:
This is even better:

ui.message("registrationapp.createdPatientMessage",ui.encodeHtml(ui.format(patient)));

Reply:
Ok that sounds good to me. I will update.

Thanks!